tower-buffer/src/lib.rs

```diff
@@ -40,6 +42,36 @@ where
     worker: worker::Handle<T::Error>,
 }
 
+/// Buffer requests with a bounded buffer
+pub struct BufferLayer {
```

Comment:
I'd recommend leaving `Buffer` out from this PR:

* We can discuss the executor question in a dedicated buffer layer PR.
* There are are changes coming to buffer (see #168).